recipes/diligent-fx/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+import os
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+import glob
+import shutil
+
+class DiligentFxConan(ConanFile):
+    name = "diligent-fx"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/DiligentGraphics/DiligentFx/"
+    description = "DiligentFX is the Diligent Engine's high-level rendering framework."
+    license = ("Apache 2.0")
+    topics = ("graphics")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], 
+    "fPIC":         [True, False],
+    }
+    default_options = {"shared": False, 
+    "fPIC": True,
+    }
+    generators = "cmake_find_package", "cmake"
+    _cmake = None
+    exports_sources = ["CMakeLists.txt", "BuildUtils.cmake", "patches/**"]
+    short_paths = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def package_id(self):
+        if self.settings.compiler == "Visual Studio":
+            if "MD" in self.settings.compiler.runtime:
+                self.info.settings.compiler.runtime = "MD/MDd"
+            else:
+                self.info.settings.compiler.runtime = "MT/MTd"
+
+    def validate(self):
+        if self.options.shared:
+            raise ConanInvalidConfiguration("Can't build as a shared lib")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def requirements(self):
+        self.requires("diligent-tools/cci.20211009")
+
+    def diligent_platform(self):
```

Comment:
```suggestion
    @property
    def _diligent_platform(self):
```

Custom methods must be protected. Also, it can be a property, as doesn`t need to change anything.